cmd/ingester/app/consumer/consumer.go

```diff
@@ -52,12 +56,18 @@ type consumerState struct {
 
 // New is a constructor for a Consumer
 func New(params Params) (*Consumer, error) {
+	var rateLimiter *time.Ticker
```

Comment:
Could you encapsulate the rate limiter in it's own struct?

Reply:
Yes